content/tutorials/quickstart/get-nftoken-list.md

```diff
@@ -0,0 +1,337 @@
+---
+html: get-nftoken-list.html
+parent: xrpl-quickstart.html
+blurb: List extant tokens on the XRP Ledger, get their data from the Clio server.
+labels:
+  - NFTokens
+  - Quickstart
+---
+
+# Get a List of NFTokens
+
+You can list all of the tokens that currently exist on the XRP Ledger. The process involves repeatedly using the `ledger_data` request, filtering the `NFToken` page objects, then listing the array of `NFToken` objects stored on the page. This example demonstrates how to extract `NFToken` objects and `NFTokenIDs`, and how to get information for the listed NFTokens using the `nft_info` request to the Clio server.
+
+## Usage
+
+You can download the [Quickstart Samples](https://github.com/XRPLF/xrpl-dev-portal/tree/master/content/_code-samples/quickstart/js/quickstart.zip){.github-code-download} archive to try the sample in your own browser.
+
+### Get NFTokens
+
+All transactions on the XRP Ledger are available for public review. You do not need an account or credentials to access information about minting and trading NFTokens.
+
+To get a list of current `NFTokens` on the XRP Ledger, click **Get NFTokens**. The button retrieves 200 transactions (the maximum batch allowance), then filters to show only the information from `NFToken` pages. The row marker for the response is stored in the **Marker** field. Each new search starts at the marker to return a unique set of records for the next 200 transactions. Click **Get NFTokens** repeatedly to retrieve all of the `NFTokens` on the XRP Ledger.
+
+### Get NFTokenIDs
+
+Click **Get NFToken IDs** for a more finely grained search that returns only the `NFTokenIDs` from each batch of 200 transactions. Continue to click **Get NFToken IDs**  to retrieve all `NFTokenIDs` on the XRP Ledger.
+
+
+### Get Info
+
+Having a list of `NFTokens` is all well and good, but it would be more interesting to do something with the `NFTokenIDs` returned. You can get information on the `NFTokens` in the response.
+
+To review `NFToken` information for a selected `NFTokenID`:
+
+1. Copy a `NFTokenID` from the list and paste it in the **NFTokenID** field.
+2. Click **Get NFToken Info**.
+
+
+# Code Walkthrough
+
+You can download the [Quickstart Samples](https://github.com/XRPLF/xrpl-dev-portal/tree/master/content/_code-samples/quickstart/js/quickstart.zip){.github-code-download} archive to try each of the samples in your own browser.
+
+## Get `NFTokens`
+
+Declare variables used in the function.
+
+```javascript
+async function getNFTokens () {
+	let jason
+	let ledger_entry_type
+	let nfTokens
+	let nftoken_data
+	let results = ""
+	let stateTree
```

Comment:
This feels like a very "C/C++" coding style, not modern JavaScript. Some of these variables (like `ledger_entry_type`) can be defined as `const` more locally when they're first assigned; only a few (like `nftoken_data`) need to be defined in advance so that they have the appropriate scope.

Also, please don't use tab characters. I notice that the zipped `ripplex8-all-nftokens.js` file mixes space and tab characters, which results in inconsistent indentation depending on how wide the user's text editor decides a tab character should be. It looks like this code was written with the assumption that a tab is two spaces wide, but one of my text editors is showing them as 4 spaces wide and the GitHub preview is showing them as 8 spaces—both of those mean that the parts that are meant to be indented the same amount don't show that way.